internal/services/qovery_services.go

```diff
@@ -0,0 +1,74 @@
+package services
+
+import (
+	"log"
+
+	"github.com/qovery/qovery-client-go"
+
+	"github.com/qovery/terraform-provider-qovery/internal/domain/credentials"
+	credsQoveryRepository "github.com/qovery/terraform-provider-qovery/internal/domain/credentials/repository/qovery"
+	credsService "github.com/qovery/terraform-provider-qovery/internal/domain/credentials/service"
+	"github.com/qovery/terraform-provider-qovery/internal/domain/organization"
+	orgaQoveryRepository "github.com/qovery/terraform-provider-qovery/internal/domain/organization/repository/qovery"
+	orgaService "github.com/qovery/terraform-provider-qovery/internal/domain/organization/service"
+)
+
+type Services struct {
+	AwsCredentialsService credentials.AwsService
+	OrganizationService   organization.Service
+}
+
+func NewQoveryServices(client *qovery.APIClient) (*Services, error) {
+	// Initializing organization service
+	orgaSvc, err := newOrganizationService(client)
+	if err != nil {
+		return nil, err
+	}
+
+	// Initializing aws cluster credentials service
+	awsCredsSvc, err := newAwsCredentialsService(client)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Services{
+		AwsCredentialsService: awsCredsSvc,
+		OrganizationService:   orgaSvc,
+	}, nil
+}
+
+func MustNewQoveryServices(client *qovery.APIClient) *Services {
+	qoveryServices, err := NewQoveryServices(client)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	return qoveryServices
+}
+
+func newOrganizationService(client *qovery.APIClient) (organization.Service, error) {
```

Comment:
As we are in the domain package, we should not have any "technical" object like the APIClient.
We should have the repository initialization outside of the domain, and then link the repository to the service.
iiuc, in the `provider.go` something like this:
```
qoveryRepository = orgaQoveryRepository.NewOrganizationQoveryRepository(client)
qoveryServices = orgaService.NewOrganizationService(qoveryRepository)
``` 
Moreover, we could name the repository implementation `qoveryRestAPI`. The "repository" term can be used only on the interface such as:
```
left_adapter --> domain <-- right_adapter

provider.go --> ( [service interface] <-- [service impl] ) --> ( [repository interface] <-- [repository impl] )

provider.go --> ( organizationService ) --> ( qoveryRestAPI )
```
Note: the `left_adapter` would be the terraform custom API, but in the code I see that `provider.go` looks like a "bootstrap"